### DISCUSSION TITLE
How to register routes?

### DISCUSSION THREAD

**Discussion by lfd-dev** [DuCanhGH/next-pwa#153] [Q&A]
I need to register a route for the share_target with post method. i.e.

```
registerRoute(
  '/share',
  shareTargetHandler,
  'POST'
);
```
However, I am unable to get this done, i have tried to do this inside custom worker also inside useEffect where I register the event.
Please assist.

**Comment by DuCanhGH** [comment]
@lfd-dev If you use GenerateSW, which you most likely do given that you attempted to use a custom worker, please use `runtimeCaching`:

```js
import withPWAInit, { runtimeCaching } from "@ducanh2912/next-pwa";

const withPWA = withPWAInit({
  runtimeCaching: [
    {
      urlPattern: "/share",
      handler: "NetworkFirst",
      method: "POST",
    },
    ...runtimeCaching,
  ],
});
```

If you want to use a custom Strategy, you are out of luck, unfortunately; your only choice is to use InjectManifest.
